File: test_data/code/python/1114-print-in-order.py
import threading
import functools

class Foo:
    def __init__(self):
        self.lock = threading.Lock()
        self.condition = threading.Condition(self.lock)
        self.done = set()

    def first(self, printFirst: 'Callable[[], None]') -> None:
        with self.lock:
            printFirst()
            self.done.add(1)

    def second(self, printSecond: 'Callable[[], None]') -> None:
        with self.lock:
            while 1 not in self.done:
                self.condition.wait()
            printSecond()
            self.done.add(2)

    def third(self, printThird: 'Callable[[], None]') -> None:
        with self.lock:
            while 2 not in self.done:
                self.condition.wait()
            printThird()
            self.done.add(3)

File: test_data/code/java/1114-print-in-order.java
class Foo {
    private boolean firstFinished, secondFinished;

    public Foo() {
        
    }

    public void first(Runnable printFirst) throws InterruptedException {
        printFirst.run();
        firstFinished = true;
    }

    public void second(Runnable printSecond) throws InterruptedException {
        while (!firstFinished) {
            Thread.sleep(1);
        }
        printSecond.run();
        secondFinished = true;
    }

    public void third(Runnable printThird) throws InterruptedException {
        while (!secondFinished) {
            Thread.sleep(1);
        }
        printThird.run();
    }
}